File: 898-transpose-matrix/transpose-matrix.java
class Solution {
    public int[][] transpose(int[][] matrix) {
        int n = matrix.length;
        int m = matrix[0].length;
        if (n != m) {
            int[][] res = new int[m][n];

            for (int i = 0; i < n; i++) {
                for (int j = 0; j < m; j++) {
                    res[j][i] = matrix[i][j];
                }
            }
            return res;
        }
        for (int i = 0; i < n; i++) {
            for (int j = i + 1; j < n; j++) {
                int temp = matrix[i][j];
                matrix[i][j] = matrix[j][i];
                matrix[j][i] = temp;
            }
        }

        return matrix;
    }
}

File: 898-transpose-matrix/transpose-matrix.cpp
class Solution {
public:
    vector<vector<int>> transpose(vector<vector<int>>& matrix) {
        int n = matrix.size();
        int m = matrix[0].size();

        if(n != m){
            vector<vector<int>> res(m, vector<int>(n));
            
            for(int i = 0; i < n; i++){
                for(int j = 0; j < m; j++){
                    res[j][i] = matrix[i][j];
                }
            }
            return res;
        }
 
        for(int i = 0; i < n; i++){
            for(int j = i + 1; j < n; j++){
                swap(matrix[i][j], matrix[j][i]);
            }
        }
        
        return matrix;
    }
};